corehq/apps/reports/views.py

```diff
@@ -1937,3 +1938,28 @@ def get_success_url(self):
     def form_valid(self, form):
         form.save()
         return super().form_valid(form)
+
+
+def get_or_create_filter_hash(request, domain):
```

Comment:
This should have some security on it like `@login_and_domain_required`. I also think that this should require a POST request rather than GET.